utils/update_checkout/update_checkout/update_checkout.py

```diff
@@ -217,7 +217,7 @@ def run_for_repo_and_each_submodule_rec(cmd):
                     shell.run(['git', 'rev-parse', '--verify', checkout_target])
                 except Exception:
                     shell.run(["git", "fetch", "--recurse-submodules=yes",
-                               "--tags"],
+                               f"{checkout_target}:{checkout_target}"],
                               echo=True, prefix=prefix)
```

Comment:
The refspec will not work with hashes or tags, and I believe it is better not to proactively create local branches if the target is a branch. I also think we still want to respect `skip_tags` here.

Reply:
Hmm yeah, this bit turned out to be trickier than I expected... The method for fetching a single tag is different from fetching a single branch, but it doesn't seem like the script knows whether `checkout_target` is a tag or a branch. Since this fetch is explicitly done because `checkout_target` wasn't found locally, it seems like simply skipping `--tags` here when `--skip-tags` is passed will lead to more errors than not. So it'd be good if we could fetch a just a single tag when needed.